department-of-veterans-affairs/abd-vro
# #2098 CancelClaim endpoint in svc-bip-api

## What was the problem?
svc-bip-api requires a cancelClaim endpoint to support partner teams.

Associated tickets or Slack threads:
- #2098 

## How does this fix it?[^1]
RabbitMqController receives a BipCloseClaimPayload, which consists of a claimId and a BipCloseClaimReason object. Controller then unpacks the previously mentioned request properties to invoke the cancelClaim method that has been added to the BipApiService. Following other BipApiService calls, the implementation of cancelClaim will formulate the bipapi url, and call the put request using the RestTemplate object.


## How to test this PR
- Positive and Negative assertions are made in the BipApiServiceTest, which use a mock httpservice object to pass explicitly configured good and bad requests.  RabbitMqIntegrationTest has also been updated with an integration test that pushes a BipCloseClaimPayload into RabbitMQ and depends on the MockBipClaimsAPI service to be up and running to receive the payload and respond appropriately.  
- Step 2[^secrel]


[^1]: [Pull-Requests guidelines](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Pull-Requests). If PR is significant, update [Current Software State](https://github.com/department-of-veterans-affairs/abd-vro/wiki/Current-Software-State) wiki page.
[^secrel]: To check if a PR will succeed in the SecRel workflow, [test PRs in the SecRel pipeline](https://github.com/department-of-veterans-affairs/abd-vro-internal/wiki/Secure-Release-process#to-test-prs-in-the-secrel-pipeline).
